tools/acorn-optimizer.mjs

```diff
@@ -1930,6 +1934,53 @@ function minifyGlobals(ast) {
   suffix = '// EXTRA_INFO:' + JSON.stringify(json);
 }
 
+function macroSubstitution(ast) {
+  const structInfoFile = wasm64
+    ? 'struct_info_generated_wasm64.json'
+    : 'struct_info_generated.json';
+  const structInfoURL = new URL('../src/' + structInfoFile, import.meta.url);
+  // Load struct and define information.
+  const structInfo = JSON.parse(read(structInfoURL));
+  const structs = structInfo.structs;
+  const defines = structInfo.defines;
+
+  fullWalk(ast, (node) => {
+    if (node.type == 'MemberExpression') {
+      if (node.object.name == 'cDefs') {
```

Comment:
We'd probably want scoping analysis here too, to avoid issues in user code with the short name `cDefs` (which might occur as the result of bundling minified code, say).

Reply:
I was going to propose some common prefix or object so it's easy to see what will be replaced as we add more. e.g. `Preprocessor.cDefs`

Reply:
Good idea.  However I also want to keep these things short if possible.  I was thinking maybe `EM.EINVAL` ?  and putting everything directly on the `EM` object?

Reply:
`EM` sounds good, but if they're going to be replaced, why keep them as short as possible?

Reply:
For readability of the source code.   These things are used a lot, and they are also often logically or'd together so repeated long prefixed just look noisy to me.. but maybe that just me?